src/main/java/Duke.java

```diff
@@ -1,10 +1,94 @@
+import java.io.IOException;
+
 public class Duke {
+    enum Command {
+        ADD,
+        DONE,
+        SAVE,
+        DELETE,
+        LIST,
+        FIND,
+        ERROR,
+        BYE
+    }
+
+    public static Storage storage;
+    public static Ui ui;
+    public static TaskList tasks;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public String run(String input) {
+        // Introduction
+
+        Command c = Parser.parse(input);
+
+        switch (c) {
+            case ADD:
+                try {
+                    Task curr = tasks.addItem(input);
+                    return ui.addedItem(curr, tasks.getListSize());
+                } catch (DukeException ex1) {
+                    return ui.showError(ex1.getMessage());
+                }
+            case DONE:
+                try {
+                    Task curr = tasks.doneItem(input);
+                    return ui.doneItem(curr);
+                } catch (DukeException ex1) {
+                    return ui.showError(ex1.getMessage());
+                }
+            case DELETE:
+                try {
+                    Task curr = tasks.deleteItem(input);
+                    return ui.deleteItem(curr);
+                } catch (DukeException ex1) {
+                    return ui.showError(ex1.getMessage());
+                }
+            case LIST:
+                return ui.returnList(tasks.getList());
+            case SAVE:
+                try {
+                    storage.overwriteData(tasks.getList());
+                    return ui.save();
+                } catch (IOException ex1) {
+                    return ui.showError(ex1.getMessage());
+                }
+            case FIND:
+                return ui.returnList(tasks.find(input));
+            case ERROR:
+                return ui.defaultError();
+            case BYE:
+                return ui.bye();
```

Comment:
I think no indentation is needed for the "cases" after a switch clause. 

I think the current design is nice and very easy to read because you make sure of enum, etc. Another alternative approach I can think of is to make use of polymorphism.For example, you can declare Command as an abstract parent class and let its subclasses override a common execute method. I believe this should shorten your code in your main class.